src/test/policies/USDSRewardDistributorTest.t.sol

```diff
@@ -0,0 +1,405 @@
+
+        vm.prank(alice);
+        vm.expectRevert(
+            abi.encodeWithSelector(ROLESv1.ROLES_RequireRole.selector, ROLE_MERKLE_UPDATER)
+        );
+        distributor.setMerkleRoot(0, bytes32(uint256(1)));
+    }
+
+    function test_setMerkleRoot_reverts_too_early() public {
+        // Still in week 0
+        vm.prank(admin);
+        vm.expectRevert(IRewardDistributor.RewardDistributor_WeekTooEarly.selector);
+        distributor.setMerkleRoot(0, bytes32(uint256(1)));
+    }
+
+    function test_setMerkleRoot_reverts_already_set() public {
+        vm.warp(startTimestamp + WEEK_DURATION);
+
+        vm.startPrank(admin);
+        distributor.setMerkleRoot(0, bytes32(uint256(1)));
+
+        vm.expectRevert(
+            abi.encodeWithSelector(IRewardDistributor.RewardDistributor_WeekAlreadySet.selector, 0)
+        );
+        distributor.setMerkleRoot(0, bytes32(uint256(2)));
+        vm.stopPrank();
+    }
+
+    // ========== Test Claiming Logic ========== //
+
+    function test_claim_as_underlying() public {
+        uint256 amount = 100e18;
+        uint40 week = 0;
+
+        // Generate leaf and proof
+        bytes32 leaf = _generateLeaf(alice, week, amount);
+        bytes32[] memory proof = new bytes32[](0); // Single leaf tree, root is leaf
+
+        // Set root
+        vm.warp(startTimestamp + WEEK_DURATION);
+        vm.prank(admin);
+        distributor.setMerkleRoot(week, leaf);
+
+        // Prepare claim data
+        uint256[] memory claimWeeks = new uint256[](1);
+        claimWeeks[0] = week;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount;
+        bytes32[][] memory proofs = new bytes32[][](1);
+        proofs[0] = proof;
+
+        // Claim
+        vm.prank(alice);
+        distributor.claim(claimWeeks, amounts, proofs, false); // asVaultToken = false
+
+        // Verify
+        assertEq(usds.balanceOf(alice), amount);
+        assertTrue(distributor.hasClaimed(alice, week));
+    }
+
+    function test_claim_as_vault_token() public {
+        uint256 amount = 100e18;
+        uint40 week = 0;
+
+        // Generate leaf and proof
+        bytes32 leaf = _generateLeaf(alice, week, amount);
+        bytes32[] memory proof = new bytes32[](0);
+
+        // Set root
+        vm.warp(startTimestamp + WEEK_DURATION);
+        vm.prank(admin);
+        distributor.setMerkleRoot(week, leaf);
+
+        // Prepare claim data
+        uint256[] memory claimWeeks = new uint256[](1);
+        claimWeeks[0] = week;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount;
+        bytes32[][] memory proofs = new bytes32[][](1);
+        proofs[0] = proof;
+
+        // Claim
+        vm.prank(alice);
+        distributor.claim(claimWeeks, amounts, proofs, true); // asVaultToken = true
+
+        // Verify
+        assertEq(sUSDS.balanceOf(alice), sUSDS.previewWithdraw(amount));
+        assertTrue(distributor.hasClaimed(alice, week));
+    }
```

Comment:
For underlying and vault both, simulate yield accrued to the vault, and that the actual claimed amount is not greater than the claim amount (which would indicate an issue with rounding). These should be fuzz tests.